utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -1004,10 +1004,13 @@ void WriteBatchWithIndex::MultiGetFromBatchAndDB(
     assert(result == WriteBatchWithIndexInternal::Result::kMergeInProgress ||
            result == WriteBatchWithIndexInternal::Result::kNotFound);
     key_context.emplace_back(column_family, keys[i], &values[i], &statuses[i]);
-    sorted_keys.emplace_back(&key_context.back());
     merges.emplace_back(result, std::move(merge_context));
   }
 
+  for (KeyContext& key : key_context) {
+    sorted_keys.emplace_back(&key);
```

Comment:
Alternatively, we could preallocate key_context if you wanted to do just one pass.